geode-apis-compatible-with-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/set/SremDUnitTest.java

```diff
@@ -125,89 +121,143 @@ public void shouldDistributeDataAmongMultipleServers_givenMultipleClients_removi
+    Set<String> results1 = jedis.smembers(key1);
+    Set<String> results2 = jedis.smembers(key2);
 
     assertThat(results1).isEmpty();
     assertThat(results2).isEmpty();
 
   }
 
+
+  @Test
+  public void shouldDistributeDataAmongCluster_givenMultipleClientsOnDifferentServer_removingSameDataFromSameSetConcurrently() {
+
+    int redisServerPort2 = clusterStartUp.getRedisPort(2);
+    JedisCluster jedis2 =
+        new JedisCluster(new HostAndPort(LOCAL_HOST, redisServerPort), redisServerPort2);
+
+    String key = "key";
+
+    List<String> members = makeMemberList(SET_SIZE, "member1-");
+    jedis.sadd(key, members.toArray(new String[] {}));
+
+    new ConcurrentLoopingThreads(SET_SIZE,
+        (i) -> jedis.srem(key, members.get(i)),
+        (i) -> jedis2.srem(key, members.get(i))).run();
+
+    Set<String> results = jedis.smembers(key);
+
+    assertThat(results).isEmpty();
+
+    jedis2.close();
+  }
+
+
   @Test
-  public void shouldDistributeDataAmongMultipleServers_givenMultipleClientsOnSameServer_removingSameDataFromSameSetConcurrently() {
+  public void shouldDistributeDataAmongCluster_givenMultipleClientsOnSameServer_removingSameDataFromSameSetConcurrently() {
 
-    Jedis jedis1B = new Jedis(LOCAL_HOST, redisServerPort1);
-    Jedis jedis2B = new Jedis(LOCAL_HOST, redisServerPort2);
+    JedisCluster jedis2 =
+        new JedisCluster(new HostAndPort(LOCAL_HOST, redisServerPort), redisServerPort);
 
     String key = "key";
 
     List<String> members = makeMemberList(SET_SIZE, "member1-");
-    jedis3.sadd(key, members.toArray(new String[] {}));
+    jedis.sadd(key, members.toArray(new String[] {}));
 
     new ConcurrentLoopingThreads(SET_SIZE,
-        (i) -> jedis1.srem(key, members.get(i)),
-        (i) -> jedis1B.srem(key, members.get(i)),
-        (i) -> jedis2.srem(key, members.get(i)),
-        (i) -> jedis2B.srem(key, members.get(i))).run();
+        (i) -> jedis.srem(key, members.get(i)),
+        (i) -> jedis2.srem(key, members.get(i))).run();
+
+    Set<String> results = jedis.smembers(key);
+
+    assertThat(results).isEmpty();
+
+    jedis2.close();
+  }
+
+  @Test
+  public void shouldDistributeDataAmongCluster_givenMultipleClientsOnDifferentServer_removingDifferentDataFromSameSetConcurrently() {
+
+    int redisServerPort2 = clusterStartUp.getRedisPort(2);
+    JedisCluster jedis2 =
+        new JedisCluster(new HostAndPort(LOCAL_HOST, redisServerPort), redisServerPort2);
+
+    String key = "key1";
+
+    List<String> members1 = makeMemberList(SET_SIZE, "member1-");
+    List<String> members2 = makeMemberList(SET_SIZE, "member2-");
+
+    List<String> allMembers = new ArrayList<>();
+    allMembers.addAll(members1);
+    allMembers.addAll(members2);
 
-    Set<String> results = jedis3.smembers(key);
+    jedis.sadd(key, allMembers.toArray(new String[] {}));
+
+    new ConcurrentLoopingThreads(SET_SIZE,
+        (i) -> jedis.srem(key, members1.get(i)),
+        (i) -> jedis2.srem(key, members2.get(i))).run();
```

Comment:
You can just use the same `jedis` instance here instead of creating a new `jedis2`.